enhancements/network/persistent-ips-sdn-secondary-networks.md

```diff
@@ -0,0 +1,743 @@
+VM start would be faster since the controller would template the pod requesting
+the IP it was allocated.
+
+The following sequence diagrams depict the create / start VM flows - the stop /
+delete flows do not have any changes from their counterparts in
+[Workflow Description](#workflow-description).
+
+##### Create VM
+
+```mermaid
+sequenceDiagram
+
+title Create VM
+actor user
+user ->> KubeVirt: createVM(name=vmA, persistentIPs=true)
+
+KubeVirt ->> Kubernetes: createPod(name=<generated name>, owner=vmA)
+
+note over OVN-K: notices a new pod was added with an owner
+OVN-K ->> OVN-K: ip := generateIP()
+
+alt if cniConf.AllowPersistentIPs()
+  OVN-K ->> Kubernetes: persistentIPAllocation(ip, owner=vmA)
+  Kubernetes -->> OVN-K: OK
+end
+Kubernetes -->> KubeVirt: OK
+
+OVN-K ->> OVN-K: configurePodNetwork(ip)
+
+note over KubeVirt: notices pod creation associated for `vmA`
+
+KubeVirt->>KubeVirt: ips := interfaceIPsMap(\n    pod.annotations[NetworkStatusAnnotation]\n)
+
+KubeVirt->>Kubernetes: updateVMStatus(ips...)
+Kubernetes -->>KubeVirt: OK
+
+KubeVirt -->> user: OK
+```
+
+##### Start stopped VM
+```mermaid
+sequenceDiagram
+
+actor user
+
+user ->> KubeVirt: startVM(vm=vmA)
+
+KubeVirt ->> Kubernetes: createPod(<br/>name=<generated name>,<br/>owner=vmA,<br/>ips=vmA.status.ips<br/>)
+
+Kubernetes -->> KubeVirt: OK
+
+note over OVN-K: notices a new pod was added
+
+KubeVirt -->> user: OK
+```
+
+**NOTES**:
+- this approach still requires the `PersistentIP` CRD to block the
+  allocation on the IP pool - i.e. preventing other workloads from being assigned
+  that IP when the VM is stopped.
+- this approach also allows the VM user to know which persistent IPs are
+  "attached" to the VM, which can be helpful when debugging.
+
+### Drawbacks
+
+TODO
+
+## Design Details
+
+### KubeVirt related changes
+
+TODO
+
+### Integrate the `PersistentIP` CRD with OVN-Kubernetes ip pool
+
+TODO
+
+### Kubernetes Garbage Collection extended info
+
+This feature is relying on Kubernetes garbage collection to delete the
+persistent IP allocations once the Virtual Machine object is deleted.
+
+But, we actually need to assure the persistent IP allocation stays around as
+long as the running Virtual Machine (i.e. the process) is running - thus, we
+also need to ensure the pod where the VM runs was deleted. If it weren't, we
+will end up in a situation where the IP would be returned to the pool earlier
+than it should, which would lead to the same IP in two workloads at the same
+time: one of them shutting down, the other already readily processing traffic.
+
+The garbage collection mechanism will behave differently depending on the
+[DeletionPropagation](https://github.com/kubernetes/apimachinery/blob/3e2600dc79feea6cdc8a9224bc8a6a7fcfee1466/pkg/apis/meta/v1/types.go#L526)
+configured by the user on the `DeleteOptions` provided when deleting the VM.
+
+#### Deleting the VM with the Background `DeletionPropagation`
```

Comment:
nit
`Background` always appears as `background` on k8s docs and of course also on command

if lower cased please change all over

Reply:
Done.